data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/PostStateValidators.java

```diff
@@ -76,7 +79,20 @@ public PostStateValidators(final DataProvider dataProvider) {
 
   @Override
   public void handleRequest(RestApiRequest request) throws JsonProcessingException {
-    final Optional<RequestBody> requestBody = request.getOptionalRequestBody();
+    final Optional<RequestBody> requestBody;
+
+    try {
+      requestBody = request.getOptionalRequestBody();
+    } catch (RuntimeException e) {
+      final Throwable throwable = Throwables.getRootCause(e);
+      if (throwable instanceof JsonParseException) {
+        request.respondError(SC_BAD_REQUEST, throwable.getMessage());
+      } else {
+        throw e;
+      }
+      return;
+    }
```

Comment:
```suggestion
    try {
      requestBody = request.getOptionalRequestBody();
    } catch (JsonProcessingException e) {
      final Throwable throwable = Throwables.getRootCause(e);
      request.respondError(SC_BAD_REQUEST, throwable.getMessage());
      return;
    } catch (RuntimeException e) {
      throw e;
    }
```

Reply:
That would not work because the exception is a `RuntimeException`, but the root cause of the exception is a `JsonProcessingException`. So cannot just catch the `JsonProcessingException` as you have.

Reply:
I'll take a look later. Maybe there is something we can do on the getOptionalRequeatBody.